File: android/app/src/main/java/com/storybooknative/stubs/ArticleEvents.kt
package com.storybooknative.stubs

import android.util.Log
import com.facebook.react.bridge.BaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.ReadableMap

class ArticleEvents : BaseJavaModule() {
    override fun getName() = "ArticleEvents"

    @ReactMethod
    fun onArticlePress(url: String) {
        Log.d(name, "onArticlePress $url")
    }

    @ReactMethod
    fun onArticleLoaded(articleId: String, extras: ReadableMap) {
        Log.d(name, "onArticleLoaded $articleId $extras")
    }

    @ReactMethod
    fun onAuthorPress(slug: String) {
        Log.d(name, "onAuthorPress $slug")
    }

    @ReactMethod
    fun onLinkPress(url: String) {
        Log.d(name, "onLinkPress $url")
    }

    @ReactMethod
    fun onVideoPress(info: ReadableMap) {
        Log.d(name, "onVideoPress $info")
    }

    @ReactMethod
    fun onTopicPress(url: String) {
        Log.d(name, "onTopicPress $url")
    }

    @ReactMethod
    fun onCommentsPress(articleId: String, url: String) {
        Log.d(name, "onCommentsPress $articleId $url")
    }

    @ReactMethod
    fun onCommentGuidelinesPress() {
        Log.d(name, "onCommentGuidelinesPress")
    }
}

File: ios/storybooknative/ArticleEvents.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface ArticleEvents : NSObject <RCTBridgeModule>
@end

@implementation ArticleEvents

RCT_EXPORT_MODULE();

RCT_EXPORT_METHOD(onArticlePress:(NSString *) url)
{
  NSLog(@"onArticlePress %@", url);
}

RCT_EXPORT_METHOD(onArticleLoaded:(NSString *) articleId extras:(id) extras)
{
  NSLog(@"onArticleLoaded %@ %@", articleId, extras);
}

RCT_EXPORT_METHOD(onAuthorPress:(NSString *) slug)
{
  NSLog(@"onAuthorPress %@", slug);
}

RCT_EXPORT_METHOD(onLinkPress:(NSString *) url)
{
  NSLog(@"onLinkPress %@", url);
}

RCT_EXPORT_METHOD(onVideoPress:(id) videoInfo)
{
  NSLog(@"onVideoPress %@", videoInfo);
}

RCT_EXPORT_METHOD(onTopicPress:(NSString *) url)
{
  NSLog(@"onTopicPress %@", url);
}

RCT_EXPORT_METHOD(onCommentsPress:(NSString *) articleId url:(NSString *) url)
{
  NSLog(@"onCommentsPress %@ %@", articleId, url);
}

RCT_EXPORT_METHOD(onCommentGuidelinesPress)
{
  NSLog(@"onCommentGuidelinesPress");
}
@end
